packages/social-previews/src/facebook-preview/link-preview.tsx

```diff
@@ -0,0 +1,84 @@
+import { __ } from '@wordpress/i18n';
+import { useCallback, useState } from 'react';
+import { TYPE_ARTICLE } from '../constants';
+import { baseDomain, facebookTitle, facebookDescription, facebookCustomText } from './helpers';
+import FacebookPostActions from './post/actions';
+import FacebookPostHeader from './post/header';
+import FacebookPostIcon from './post/icons';
+import type { FacebookPreviewProps } from './types';
+
+const LANDSCAPE_MODE = 'landscape';
+const PORTRAIT_MODE = 'portrait';
+
+const FacebookLinkPreview: React.FC< FacebookPreviewProps > = ( {
+	url,
+	title,
+	description,
+	image,
+	user,
+	customText,
+	type,
+} ) => {
+	const [ mode, setMode ] = useState( LANDSCAPE_MODE );
+	const isArticle = type === TYPE_ARTICLE;
+	const portraitMode = ( isArticle && ! image ) || mode === PORTRAIT_MODE;
+
+	const getModeClass = useCallback(
+		() => `is-${ portraitMode ? 'portrait' : 'landscape' }`,
+		[ portraitMode ]
+	);
```

Comment:
`useCallback` is an overkill for this. a direct calculation will be much faster than `useCallback`.

```suggestion
	const modeClass = `is-${ portraitMode ? 'portrait' : 'landscape' }`;
```

Reply:
That's right!